### ISSUE TITLE
trivy image scan shows some critical results and old Ubuntu?

### ISSUE THREAD

**Issue by KomtGoed** [jacobalberty/unifi-docker#503]
### Operating system

Debian 10.11

### UniFi Tag

stable

### Docker run

docker run -it aquasec/trivy image jacobalberty/unifi

### Bug description

Scanning with
docker run -it aquasec/trivy image jacobalberty/unifi
(for Trivy see https://github.com/aquasecurity/trivy) shows some critical issues in packages contained.

Not sure if I read the results correctly (e.g. | libjpeg-turbo8            | CVE-2018-11813   |          | **1.5.2-0ubuntu5.18.04.4**  ) but it seems the underlying Ubuntu image is rather old.

Would it be possible to update the respective packages?

**Comment by jacobalberty** [comment]
Once we update past ubuntu 18.04  arm 32 support has to be dropped. This is detailed in pull request #485 .  I did evaluate building mongodb 3.2 on Ubuntu 20.04 but ran into compiling issues that weren't immediately apparent how to fix. I would gladly look at a pr that builds mongodb 3.2 or newer on ubuntu 20.04 but for right now I don't have the time to get that cross compile working to maintain arm 32 support and I'd rather push killing arm 32 support out as far as I can. Ubuntu 18.04 doesn't hit EOL till April 2023. In fact it's running on ubuntu 18.04.6:
```
 docker run --rm -it jacobalberty/unifi:latest cat /etc/lsb-release
[2021-12-17 20:09:28,105] <docker-entrypoint> Executing: cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.6 LTS"
```
18.04.6 was released on [September 17th 2021](https://lists.ubuntu.com/archives/ubuntu-announce/2021-September/000272.html) so it's about 2 months old I wouldn't call the underlying image all that old. However [ubuntu does not have a fix for it available for any 18.04 version](https://ubuntu.com/security/CVE-2018-11813). 

As for [CVE-2018-11813](https://nvd.nist.gov/vuln/detail/CVE-2018-11813) It's a denial of service vulnerability, not a code execution vulnerability, that would require libjpeg to be provided with user input to trigger. As far as I can tell there is no way to provide user input to libjpeg without being logged in and using the functionality that lets you upload a map image. In addition to that [redhat's reporting on the vulnerability](https://access.redhat.com/security/cve/cve-2018-11813) indicates it's not actually a vulnerability in libjpeg, rather it's a vulnerability in the "cjpeg" tool that comes with libjpeg. which means in order to actually exploit it an attacker would have to gain some sort of remote access to the container first. I think it's safe to say CVE-2018-11813 is ok to ignore. I do have docker security scanning on the image enabled.

There are of course other CVE's listed.

I would discount anything under the Java (jar) section as it is either other jar files that were installed with java but not used by unifi, or part of the core unifi and really up to unifi to provide fixes for their current stable tag. For critical fixes like log4shell I am open to either pull requests to provide a hot fix as outlined in the [hotfixes README.md](../hotfixes/README.md) to fix the current version, or for fixes that are done in the current version but not in `stable-5` or `v6.0.45` I am open to applying hotfix pull requests to those.

for other CVE's just looking at the output of tools like trivy can be deceptive (see my explanation of the libjpeg vulnerability you brought up). I can address them individually if you have specific concerns about a tool, and will do my best to provide a fix (or merge in a relevant pull request to fix it) if there is a critical vulnerability.  Outside of that We could try and slim down some of the dependencies that either came default with the image or were added as part of the "build" process but missed in the cleanup.

I'm not really seeing much in terms of build dependencies that can be removed but maybe some of the base utilities could be trimmed. We could try doing some workarounds to update openjdk-8-jre-headless to openjdk-11-jre-headless and that would reduce some of the vulnerabilities listed as well as an attack surface reduction, but just swapping out the packages won't work. I'm taking a look at the feasibility of that right now.

**Comment by KomtGoed** [comment]
"looking at the output of tools like trivy can be deceptive"... I was already "afraid" that reality is more complex. Thanks for looking into this so quickly and thoroughly! Glad to hear that security is so clearly on your radar. Sounds like it is in safe hands and I can better focus on things I know more off :)
